flow/src/test/scala/org/alephium/flow/core/BlockFlowGroupViewSpec.scala

```diff
@@ -20,13 +20,21 @@ import scala.collection.mutable
 
 import akka.util.ByteString
 
-import org.alephium.flow.FlowFixture
+import org.alephium.flow.{AlephiumFlowSpec, FlowFixture}
 import org.alephium.protocol.ALPH
-import org.alephium.protocol.model.{AssetOutput, ChainIndex, GroupIndex}
-import org.alephium.util.{AlephiumSpec, AVector, TimeStamp}
+import org.alephium.protocol.Generators.hashGen
+import org.alephium.protocol.model.{
+  AssetOutput,
+  ChainIndex,
+  ContractOutputRef,
+  GroupIndex,
+  Hint,
+  TxOutputRef
+}
+import org.alephium.util.{AVector, TimeStamp}
 
-class BlockFlowGroupViewSpec extends AlephiumSpec {
-  it should "fetch preOutputs" in new FlowFixture {
+class BlockFlowGroupViewSpec extends AlephiumFlowSpec {
+  it should "fetch getPreAssetOutputs and getPreOutputs" in new FlowFixture {
```

Comment:
It seems getPreContractOutput is tested instead of getPreOutputs?

Reply:
There were preOutputs tests already but only for `additionalCache`, adding the missing case + moving `getPreContractOutput` for missing contractRef to a better place, you are right :+1: